### DISCUSSION TITLE
Installation error: Slot is too small for image

### DISCUSSION THREAD

**Discussion by anthonio9** [rauc/rauc#1570] [Q&A]
After adding a heavy package to the image, the RAUC bundle and the new slot size are greatly larger than the the existing slot and an error like below occurred during the `rauc install` operation:

```sh
100% Installing failed.
LastError: Installation error: Failed updating slot rootfs.1: Slot (693518336 bytes) is too small for image (812767232 bytes).
idle
Installing `/tmp/update-bundle.raucb` failed
```

I'm aware that RAUC has some resizing capabilities, however I'm not exactly sure how to apply them. There is a way to enable resizing via `resize=true` in *system.conf* - that does not help at all sadly. Are there any other way of fixing this issue? Packages `parted`, `resizefs` are present on my embedded system.

**Comment by ejoerns** [comment]
@anthonio9 The `resize` option is for (ext4) file systems only and will resize the *file system* to span the entire slot (*after* copying).

Your issue is that the *slot* is too small for the image. This can only be fixed by making the image smaller or repartitioning your disk.

**Reply by anthonio9** [reply]
I see, do I guess right that to make the slot larger a fixed-size would have to be set in the wic recipe? Currently I'm having below:
```sh
part u-boot --source rawcopy --sourceparams="file=imx-boot" --ondisk mmcblk --no-table --align ${IMX_BOOT_SEEK}
part /boot --source bootimg-partition --ondisk mmcblk --fstype=vfat --label boot --active --align 8192 --size 64
part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_A --align 8192
part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_B --align 8192
part /data --fixed-size 100M --ondisk mmcblk --fstype=ext4 --label data --align 8192
```
My guess is that applying `--fixed-size 1G` for both rootfs partitions would make things work better again.

**Reply by anthonio9** [reply]
Fixing the size of the partitions at a larger value did help.

For now I'll be happy with either increasing the partition size with WIC or decreasing the update slot size. Thanks for the help.

**Comment by jluebbe** [comment]
The `resize=true` option grows the *filesystem* to the size of the *partition*, so it won't help in the case where the *partition* is too small.

RAUC doesn't change the partition table, as layout changes can't be done atomically (at least in a general enough way to support them in an updater).

One possible approach would be to add a resizing mechanism to the initramfs: before mounting the rootfs, you can do everything that's needed to repartition both rootfs slots. If you do that carefully enough, that could be done without risking bricking the device due to power failure. In a follow-update, you'd then be able to use the larger partition size to add your heavy package.

**Reply by anthonio9** [reply]
Would a service such as `rauc-grow-data-part` used in [meta-rauc-community](https://github.com/rauc/meta-rauc-community/blob/scarthgap/meta-rauc-nxp/recipes-core/rauc/rauc_%25.bbappend) help in my case?

**Reply by jluebbe** [reply]
The [script that it calls](https://github.com/rauc/meta-rauc-community/blob/scarthgap/meta-rauc-nxp/recipes-core/rauc/files/grow-data-partition.sh) is a bit similar to what you'd need to do in an initramfs, if you'd need to resize the rootfs in the field. In practice it would be trickier though. The data partition is easier because it can just be unmounted.  :)
